### PR TITLE
Add tracking to code copy

### PR DESCRIPTION
**Pull request by hartantothio**
### Description

Send copy event to Segment.

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/track-copy-code

### CODE REVIEW

File: assets/js/segment.js [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+      const selection = getSelection();
+      selection.removeAllRanges();
+
+      const range = document.createRange();
+      range.selectNodeContents(node);
+      selection.addRange(range);
+      copyText = range.toString();
+
+      document.execCommand('copy');
+      selection.removeAllRanges();
```
Comment (rafikalid) [inline assets/js/segment.js:19]: Maybe you need to restore selected ranges from line 11 no?

File: assets/js/segment.js [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
 
   // Determine whether we should track or not
   function shouldTrack(e) {
-    if (e.target.tagName === 'A' || isTargetCodeCopyButton(e.target)) {
+    if (e.target.tagName === 'A') {
       return { target: e.target };
-    } else if (e.target.parentNode.tagName === 'A') {
+    } else if (
+      isTargetCodeCopyButton(e.target) ||
+      e.target.parentNode.tagName === 'A'
```
Comment (rafikalid) [inline assets/js/segment.js:36]: Maybe not just parent,  we should check all top parents 
I suggest use:
`if(e.target.closest('a')){ ... }`

Comment (hartantothio) [reply assets/js/segment.js:36]: I looked at the structure for the site, and it's pretty straightforward. I think using `closest` (it probably has some perf impact) could certainly work, but I'll have to refactor more to accommodate that use case. I'll address that at a separate point.

**Review by danshafie — APPROVED** [review]

**Review by rafikalid — APPROVED** [review]

**Review by bbejeck — APPROVED** [review]
piggybacking on @danshafie and @rafikalid approval

**Pull request merged**